.github/workflows/cross-version-tests.yml

```diff
@@ -55,18 +55,21 @@ jobs:
           # https://github.com/actions/virtual-environments/blob/main/images/linux/Ubuntu2004-README.md#java
           java-version: 8
           distribution: 'adopt'
-      - name: Enable conda
-        run: |
-          echo "/usr/share/miniconda/bin" >> $GITHUB_PATH
-      - name: Update python
+      - name: Get python version
+        id: get-python-version
         run: |
           if [[ "${{ matrix.package }}" = "scikit-learn" && "${{ matrix.version }}" = "dev" ]]; then
             python_version=3.7
           else
             python_version=3.6
           fi
-          conda install python=$python_version
-          python --version
+          echo "::set-output name=python-version::$python_version"
+      - uses: actions/setup-python@v2
+        with:
+          python-version: ${{ steps.get-python-version.outputs.python-version }}
```

Comment:
The following python versions are pre-installed in the ubuntu 20.04 image for GitHub Actions:

https://github.com/actions/virtual-environments/blob/main/images/linux/Ubuntu2004-README.md#python

```
2.7.18
3.5.10
3.6.13
3.7.10
3.8.10
3.9.5
```

If we use one of these versions,  what `actions/setup-python@v2` does is just switch the python version, which should be pretty stable.